gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1419,10 +1419,15 @@ public DiskStoreFactory createDiskStoreFactory(DiskStoreAttributes attrs) {
   // this snapshot is different from snapshot for export.
   // however this can be used for that purpose.
   public boolean snapshotEnabled() {
-    return !DEFAULT_SNAPSHOT_DISABLED;
+    // if rowstore return false
+    // if snappy return true
+    if (getInternalProductCallbacks().isSnappyStore()) {
+      return !DEFAULT_SNAPSHOT_ENABLED;
```

Comment:
This condition looks incorrect. You will disable snapshot for SnappyData when user has explicitly enabled it?

This should rather go into the initialization of DEFAULT_SNAPSHOT_ENABLED itself and this method should simply return DEFAULT_SNAPSHOT_ENABLED:

```java
private final boolean DEFAULT_SNAPSHOT_ENABLED = SystemProperties.getServerInstance().getBoolean(
    "cache.ENABLE_DEFAULT_SNAPSHOT_ISOLATION", getInternalProductCallbacks().isSnappyStore() /* defaultValue */);
```